src/lib/sort-modifiers.ts

```diff
@@ -6,33 +6,67 @@ import { AnyConfig } from './types'
  * - When an arbitrary variant appears, it must be preserved which modifiers are before and after it
  */
 export const createSortModifiers = (config: AnyConfig) => {
-    const orderSensitiveModifiers = Object.fromEntries(
-        config.orderSensitiveModifiers.map((modifier) => [modifier, true]),
-    )
+    // Pre-compute weights for all known modifiers for O(1) comparison
+    const modifierWeights = new Map<string, number>()
 
-    const sortModifiers = (modifiers: string[]) => {
-        if (modifiers.length <= 1) {
-            return modifiers
-        }
+    // Assign weights to sensitive modifiers (highest priority, but preserve order)
+    config.orderSensitiveModifiers.forEach((mod, index) => {
+        modifierWeights.set(mod, 1000000 + index) // High weights for sensitive mods
+    })
+
+    return (modifiers: readonly string[]): string[] => {
+        // Fast path for common cases
+        if (modifiers.length <= 1) return [...modifiers]
+
+        const result: string[] = []
+        let currentSegment: string[] = []
 
-        const sortedModifiers: string[] = []
-        let unsortedModifiers: string[] = []
+        // Process modifiers in one pass
+        for (let i = 0; i < modifiers.length; i++) {
+            const modifier = modifiers[i]!
 
-        modifiers.forEach((modifier) => {
-            const isPositionSensitive = modifier[0] === '[' || orderSensitiveModifiers[modifier]
+            // Check if modifier is sensitive (starts with '[' or in orderSensitiveModifiers)
+            const isArbitrary = modifier[0] === '['
+            const isOrderSensitive = modifierWeights.has(modifier)
 
-            if (isPositionSensitive) {
-                sortedModifiers.push(...unsortedModifiers.sort(), modifier)
-                unsortedModifiers = []
+            if (isArbitrary || isOrderSensitive) {
+                // Sort and flush current segment using numeric weights
+                if (currentSegment.length > 0) {
+                    currentSegment.sort((a, b) => {
+                        const weightA = modifierWeights.get(a)
+                        const weightB = modifierWeights.get(b)
+                        if (weightA !== undefined && weightB !== undefined) {
+                            return weightA - weightB
+                        }
+                        if (weightA !== undefined) return -1
+                        if (weightB !== undefined) return 1
+                        return a.localeCompare(b)
+                    })
```

Comment:
I don't understand what the sorting here is for. `weightA` and `weightB` will always be undefined here since only modifiers that are not order-sensitive will be pushed into `currentSegment`. The order-sensitive modifiers also don't need to be sorted against each other, they just split the segments which themselves need to be sorted.

Looks to me that the changes here are not necessary. (Or am I missing something?) Could you remove them?